sim/core/apl_actions_casting.go

```diff
@@ -313,7 +313,18 @@ func (action *APLActionCastAllStatBuffCooldowns) Execute(sim *Simulation) {
 	}
 }
 func (action *APLActionCastAllStatBuffCooldowns) String() string {
-	return fmt.Sprintf("Cast All Buff Cooldowns For: %s", StringFromStatTypes(action.statTypesToMatch))
+	return fmt.Sprintf("CastAllBuffCooldownsFor(%s)", StringFromStatTypes(action.statTypesToMatch))
+}
+func (action *APLActionCastAllStatBuffCooldowns) PostFinalize(rot *APLRotation) {
+	if len(action.allSubactions) == 0 {
+		rot.ValidationWarning("%s will not cast any spells! There are either no major cooldowns buffing the specified stat type(s), or all of them are manually cast in the APL.", action)
+	} else {
+		actionIDs := MapSlice(action.allSubactions, func(subaction *APLActionCastSpell) ActionID {
+			return subaction.spell.ActionID
+		})
+
+		rot.ValidationWarning("%s will cast the following spells: %s", action, StringFromActionIDs(actionIDs))
```

Comment:
Could we not leave in the action IDs and then in frontend map them to actual spells with icons? Like we do in the logs parser? Could maybe be cleaner/easier to visually reference for users?
Or perhaps list them on a new line?

Reply:
It should already be printing the exact same string as it does for single spells, just separated by commas. Is there something I need to hook into in the UI code to pretty-print it? It's already recognizing the ActionID string automatically and cleanly displaying just the spell name, just without the icon etc.

Reply:
Yea, in the UI we would probably need to add a regex that parses the validation message > makes an ActionID out of its > replaces it with a `<a href>` and initialize the wowhead tooltips code. But that can also be a future state if needed. 
Actually, I found it already being used in the APL here: https://github.com/wowsims/cata/blob/4913dd7a9e209b427ee1acab40e8bd45eb6eb8f3/ui/core/components/individual_sim_ui/apl_rotation_picker.ts#L266

Reply:
Yeah it's clearly already doing some parsing automatically, or else it wouldn't be printing the spell names cleanly (the direct %s of an ActionID struct does not look like that)

Reply:
Ohhhh, my bad then 👍 feel free to merge regardless